instrumentation/github.com/confluentinc/confluent-kafka-go/kafka/splunkkafka/test/kafka_test.go

```diff
@@ -47,7 +46,54 @@ var (
 	testTopic   = "gotest"
 )
 
-func TestSynchronous(t *testing.T) {
+func TestChannelBasedProducer(t *testing.T) {
+	defer goleak.VerifyNone(t)
+
+	partition := int32(0)
+	sr, opts := newFixtures()
+	p := newProducer(t, opts...)
+
+	done := make(chan struct{})
+	var sent *kafka.Message
+	go func() {
+		defer close(done)
+		sent = requireEventIsMessage(t, <-p.Events())
+	}()
+	go func() {
+		p.ProduceChannel() <- &kafka.Message{
+			TopicPartition: kafka.TopicPartition{
+				Topic:     &testTopic,
+				Partition: partition,
+			},
+			Key:   key,
+			Value: val,
+		}
+	}()
+
+	// Wait for the delivery report goroutine to finish.
+	<-done
+	require.NoError(t, sent.TopicPartition.Error)
+
+	p.Close()
```

Comment:
is it imortant that this `p.Close()` is here?
maybe we could `defer p.Close()` after `p := newProducer(t, opts...)`?

same for `TestFunctionBasedProducer`

Reply:
`Close` flushes any state held by the underlying `Producer`/`Consumer`. Having it called before we evaluate any spans ensures that all spans are going to be accounted for, both now and if any others are added in the future.

It seems like a good idea to keep the call where it is for these reasons, no?

Reply:
You are totally right 👍 Forgot about that